In [3]:
%pip install kagglehub
%pip install pandas

  Using cached kagglehub-0.3.4-py3-none-any.whl.metadata (22 kB)
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
  Using cached tqdm-4.67.0-py3-none-any.whl.metadata (57 kB)
  Using cached charset_normalizer-3.4.0-cp312-cp312-macosx_11_0_arm64.whl.metadata (34 kB)
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
  Using cached urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
  Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
Using cached kagglehub-0.3.4-py3-none-any.whl (43 kB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached tqdm-4.67.0-py3-none-any.whl (78 kB)
Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
Using cached charset_normalizer-3.4.0-cp312-cp312-macosx_11_0_arm64.whl (119 kB)
Using cached idna-3.10-py3-none-any.whl (70 kB)
Using cached urllib3-2.2.3-py3-none-any.whl (126 kB)
Note: you may need to restart the kernel to use updated packages.
  Using cached pandas-2.2.3-cp312-cp312-macosx_11_0_arm64.whl.m

In [6]:
import kagglehub
import os
import pandas as pd

# Téléchargement du dataset avec kagglehub
path = kagglehub.dataset_download("prathamsaraf1389/spotify-global-top-50-daily-update")

print("SPOTIFYYYYYY :", path)

# Remplace "playlist.csv" par le nom exact trouvé
file_path = os.path.join(path, "playlist.csv")
spotify_data = pd.read_csv(file_path)

# Affiche un aperçu des données
print(spotify_data.head())


SPOTIFYYYYYY : /Users/jbb/.cache/kagglehub/datasets/prathamsaraf1389/spotify-global-top-50-daily-update/versions/315
           track_name track_add_date track_add_time  multiple_artists_bool  \
0             Flowers     2023-03-31       10:02:16                  False   
1           Kill Bill     2023-03-31       10:02:16                  False   
2                 TQG     2023-03-31       10:02:16                   True   
3                BESO     2023-03-31       10:02:16                   True   
4  Boy's a liar Pt. 2     2023-03-31       10:02:16                   True   

                   name_of_artists               album_name  \
0                  ['Miley Cyrus']  Endless Summer Vacation   
1                          ['SZA']                      SOS   
2           ['KAROL G', 'Shakira']       MAÑANA SERÁ BONITO   
3    ['ROSALÍA', 'Rauw Alejandro']                       RR   
4  ['PinkPantheress', 'Ice Spice']       Boy's a liar Pt. 2   

  album_release_date album_release_

# Introduction

Fondée en 2006 par Daniel Ek & Martin Lorentzon en Suède, Spotify est devenue l'une des plateformes de streaming musical les plus populaires au monde, révolutionnant la manière d'écouter la musique en offrant un accès instantané à un vaste cataloque de morceaux, albums et podcast.

La plateforme propose une multitude de playlists dont la célèbre "Top 50 Global", une sélection des morceaux les plus écoutés dans le monde entier, cette playlist, étant mise à jour quotidiennement, reflète les goûts musicaux globaux en temps réel en se basant sur le nombre de streams, dont artistes, producteurs et labels, voient une vitrine essentielle pour mesure le succès de leurs titre et toucher une audience mondiale.

# Présentation de la data

Nous avons les lignes suivantes dans notre database

track_name: Le nom du morceau

track_add_date: La date de l'ajout du morceau dans la playlist

track_add_time: L'heure de l'ajout du morceau dans la playlist

multiple_artists_bool: Un booleen qui nous renseigne si le morceau est mono ou multi artiste

name_of_artists: Nom des artistes

album_name: Nom de l'album

album_release_date: Date de sortie de l'album

album_release_date_precision: Précision de la date de sortie de l'album

number_of_tracks_in_album: Nombre de morceaux dans l'album

position_in_playlist: Position du morceau de la playlist

track_duration_ms: Durée du morceau en milisecondes

track_popularity: Popularité du morceau 

track_explicit: Un booleen qui nous donne si le morceau est explicite ou non

images_path: Lien pour image du morceau


# Premières observations

In [7]:
spotify_data.describe()

number_of_tracks_in_album  position_in_playlist  track_duration_ms  \
count               15550.000000          15550.000000       15550.000000   
mean                   10.519807             25.500000      196956.244437   
std                     9.213601             14.431334       38362.990747   
min                     1.000000              1.000000       36226.000000   
25%                     1.000000             13.000000      172797.000000   
50%                    12.000000             25.500000      195986.000000   
75%                    17.000000             38.000000      222461.000000   
max                   119.000000             50.000000      405906.000000   

       track_popularity  
count      15550.000000  
mean          89.535691  
std           11.933126  
min            0.000000  
25%           88.000000  
50%           92.000000  
75%           95.000000  
max          100.000000

Nous avons 15550 morceaux dans ce dataset, cela correspond à 15550/50 = 311 semaines de Globat Top 50

On va pouvoir observer quels sont les morceaux les plus récurrents, le nombre d'artistes différents et la distribution de la durée des morceaux.

In [8]:
# Compter le nombre d'occurrences de chaque artiste
artist_counts = spotify_data['name_of_artists'].value_counts()

# Obtenir les 10 artistes les plus présents
top_10_artists = artist_counts.head(10)

print(top_10_artists)

name_of_artists
['Taylor Swift']               1077
['Olivia Rodrigo']              348
['SZA']                         340
['Miley Cyrus']                 331
['Arctic Monkeys']              308
['Harry Styles']                269
['David Kushner']               255
['Doja Cat']                    252
['Bad Bunny']                   242
['The Weeknd', 'Daft Punk']     242
Name: count, dtype: int64


In [11]:
# Filter the data for the top 10 artists
top_10_artists_data = spotify_data[spotify_data['name_of_artists'].apply(lambda x: any(artist in x for artist in top_10_artists.index))]

# Count the number of unique tracks for each of the top 10 artists
unique_tracks_per_artist = top_10_artists_data.groupby('name_of_artists')['track_name'].nunique().sort_values(ascending=False)

print(unique_tracks_per_artist)

name_of_artists
['Taylor Swift']               53
['Olivia Rodrigo']             15
['Bad Bunny']                   5
['Doja Cat']                    3
['Miley Cyrus']                 3
['SZA']                         3
['Arctic Monkeys']              1
['David Kushner']               1
['Harry Styles']                1
['The Weeknd', 'Daft Punk']     1
Name: track_name, dtype: int64


In [10]:
# Get the top track for each of the top 10 artists based on track popularity
top_tracks = top_10_artists_data.loc[top_10_artists_data.groupby('name_of_artists')['track_popularity'].idxmax()]

# Display the top tracks
print(top_tracks[['name_of_artists', 'track_name', 'track_popularity']])

                  name_of_artists          track_name  track_popularity
4720           ['Arctic Monkeys']    I Wanna Be Yours                96
4921                ['Bad Bunny']      WHERE SHE GOES               100
4858            ['David Kushner']            Daylight                99
8350                 ['Doja Cat']  Paint The Town Red               100
9474             ['Harry Styles']           As It Was                95
4628              ['Miley Cyrus']     Angels Like You                95
5906           ['Olivia Rodrigo']             vampire                99
52                        ['SZA']           Kill Bill                95
5453             ['Taylor Swift']        Cruel Summer               100
1078  ['The Weeknd', 'Daft Punk']             Starboy                94


In [ ]:
# Count the occurrences of each track
track_counts = spotify_data['track_name'].value_counts()

# Get the 10 most present tracks
top_10_tracks = track_counts.head(10)

print(top_10_tracks)

track_name
I Wanna Be Yours    308
Kill Bill           273
As It Was           269
Daylight            255
Cruel Summer        250
Starboy             242
Flowers             233
Die For You         204
Anti-Hero           199
Classy 101          187
Name: count, dtype: int64


In [ ]:
# Count the occurrences of each track with its artist(s)
track_artist_counts = spotify_data.groupby(['track_name', 'name_of_artists']).size().reset_index(name='count') # nolint

# Get the 10 tracks that spent the most time in the dataset
top_10_tracks_with_artists = track_artist_counts.nlargest(10, 'count')

print(top_10_tracks_with_artists)

track_name
Seven (feat. Latto) (Explicit Ver.)    67
Ella Baila Sola                        51
greedy                                 33
All I Want for Christmas Is You        27
Paint The Town Red                     23
Name: count, dtype: int64
